std/bitmanip.d

```diff
@@ -1513,6 +1653,67 @@ public:
         }
     }
 
+    /**
+     * Creates `BitArray` from a string.
+     * Params:
+     *   s = string consisting of `0`, `1` and separators `_`.
+     * Throws:
+     *   `Exception` if the given string contains invalid characters.
+     */
+    static BitArray fromString(string s) pure @system
+    {
+        import std.exception : enforce;
+        import std.string : lastIndexOfNeither;
+        import std.algorithm.searching : count;
+
+        enforce(s.lastIndexOfNeither("01_") == -1,
```

Comment:
Actually, remove this and make the default case of the switch I mention in the other comment throw. That way you're only looping over the string twice instead of three times.